web/components/widgets/pagination.tsx

```diff
@@ -35,56 +36,124 @@ export function PaginationNextPrev(props: {
+        'mx-auto flex w-full items-center gap-4 bg-white pt-2 pb-6',
         className
       )}
       aria-label="Pagination"
     >
-      <div className="hidden sm:block">
-        <p className="text-sm text-gray-700">
-          Showing <span className="font-medium">{page * itemsPerPage + 1}</span>{' '}
-          to{' '}
-          <span className="font-medium">
-            {Math.min(totalItems, (page + 1) * itemsPerPage)}
-          </span>{' '}
-          of <span className="font-medium">{totalItems}</span> results
-        </p>
-      </div>
-      <PaginationNextPrev
-        prev={page > 0 ? prevTitle ?? 'Previous' : null}
-        next={page < maxPage ? nextTitle ?? 'Next' : null}
-        onClickPrev={() => setPage(page - 1)}
-        onClickNext={() => setPage(page + 1)}
+      <PaginationArrow
+        scrollToTop={scrollToTop}
+        onClick={() => setPage(page - 1)}
+        disabled={page <= 0}
+        nextOrPrev="prev"
+      />
+      <Row className="gap-2">
+        {pageNumbers.map((pageNumber) => (
+          <PageNumbers pageNumber={pageNumber} setPage={setPage} page={page} />
+        ))}
+      </Row>
+      <PaginationArrow
         scrollToTop={scrollToTop}
+        onClick={() => setPage(page + 1)}
+        disabled={page >= maxPage}
+        nextOrPrev="next"
       />
     </nav>
   )
 }
+
+export function PaginationArrow(props: {
+  scrollToTop?: boolean
+  onClick: () => void
+  disabled: boolean
+  nextOrPrev: 'next' | 'prev'
+}) {
+  const { scrollToTop, onClick, disabled, nextOrPrev } = props
+  return (
+    <a
+      href={scrollToTop ? '#' : undefined}
+      onClick={onClick}
+      className={clsx(
+        'rounded-lg transition-colors',
+        disabled
+          ? 'text-greyscale-2 pointer-events-none'
+          : 'hover:bg-greyscale-1.5 cursor-pointer text-indigo-700'
+      )}
+    >
+      {nextOrPrev === 'prev' && (
+        <ChevronLeftIcon className="h-[24px] w-[24px]" />
+      )}
+      {nextOrPrev === 'next' && (
+        <ChevronRightIcon className="h-[24px] w-[24px]" />
+      )}
+    </a>
+  )
+}
+
+export function PageNumbers(props: {
+  pageNumber: pageNumbers
+  setPage: (page: number) => void
+  page: number
+}) {
+  const { pageNumber, setPage, page } = props
+  if (pageNumber === '...') {
+    return <div className="text-greyscale-4 select-none">...</div>
+  }
+  return (
+    <button
+      onClick={() => setPage(pageNumber)}
+      className={clsx(
+        'select-none rounded-lg px-2',
+        page === pageNumber
+          ? 'bg-indigo-100 text-indigo-700'
+          : 'text-greyscale-4 hover:bg-greyscale-1.5'
+      )}
+    >
+      {pageNumber + 1}
+    </button>
+  )
+}
+
+type pageNumbers = number | '...'
+
+export function getPageNumbers(
+  maxPage: number,
+  page: number
+): Array<pageNumbers> {
+  if (maxPage <= 7) {
+    return Array.from({ length: maxPage + 1 }, (_, index) => index)
```

Comment:
This works. You could also use `range(0, maxPage + 1)` from lodash

Ditto below.

Reply:
I think there's some type issues with that cause it could be strings or numbers